dao/src/main/java/com/epam/esm/dao/converter/impl/GiftCertificateConverter.java

```diff
@@ -0,0 +1,66 @@
+package com.epam.esm.dao.converter.impl;
+
+import com.epam.esm.dao.converter.Converter;
+import com.epam.esm.dao.entity.GiftCertificate;
+import com.epam.esm.model.entity.GiftCertificateDto;
+import com.epam.esm.model.entity.TagDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class GiftCertificateConverter implements Converter<GiftCertificate, GiftCertificateDto> {
+
+    private final TagConverter tagConverter;
+
+    @Autowired
+    public GiftCertificateConverter(TagConverter tagConverter) {
+        this.tagConverter = tagConverter;
+    }
+
+    public GiftCertificateDto convertTo(GiftCertificate entity) {
+        GiftCertificateDto giftCertificateDto = new GiftCertificateDto();
+        if (entity.getId() !=0) {
+            giftCertificateDto.setId(entity.getId());
```

Comment:
Think GiftCertificateDto.id field should be `Long`, so it has `null` as a default non-present value instead of `0`

Reply:
done